ui_testing/pages/header_page.py

```diff
@@ -220,6 +222,177 @@ def click_on_clear_all(self):
+
+    def click_on_roles_permissions_button(self):
+        self.base_selenium.LOGGER.info('Press on roles & permissions button')
+        self.base_selenium.click(element='header:roles_and_permissions_button')
+        self.sleep_small()
+
+    def click_create_new_role(self):
+        self.base_selenium.LOGGER.info('Press on create new role button')
+        self.base_selenium.click(element='roles_and_permissions:new_role_btn')
+        self.sleep_small()
+
+    def get_roles_page(self):
+            self.base_selenium.LOGGER.info(' + Get roles page.')
+            self.base_selenium.get(url=self.role_url)
+            self.sleep_small()
+
+    def click_on_role_overview_btn(self):
+        self.base_selenium.LOGGER.info('Press on the overview  button')
+        self.base_selenium.click(element='roles_and_permissions:roles_overview_btn')
+        self.sleep_small()
+
+    def set_role_name(self, role_name):
+
+            self.base_selenium.set_text(element='roles_and_permissions:role_name', value=role_name)
+
+    def get_role_name(self):
+            return self.base_selenium.get_text(element='roles_and_permissions:role_name').split('\n')[0]
+
+    def create_new_role(self, sleep=True, role_name = ''):
+        self.base_selenium.LOGGER.info(' + Create new role.')
+        self.base_selenium.click(element='roles_and_permissions:new_role_btn')
+        time.sleep(self.base_selenium.TIME_SMALL)
+        role_name = self.set_role_name(role_name)
+
+        role_data = {
+            "role_name":role_name,
+        }
+        self.save(sleep)
+        return role_data
+
+    def archive_selected_roles(self):
+            self.base_selenium.scroll()
+            self.base_selenium.click(element='roles_and_permissions:right_menu')
+            self.base_selenium.click(element='roles_and_permissions:archive')
+            self.confirm_popup()
+
+    def get_archived_roles(self):
+        self.base_selenium.scroll()
+        self.base_selenium.click(element='roles_and_permissions:right_menu')
+        self.base_selenium.click(element='roles_and_permissions:archived')
+        self.sleep_small()
+
+    def click_on_role_right_menu(self):
+        self.base_selenium.LOGGER.info('Press on the right menu')
+        self.base_selenium.click(element='roles_and_permissions:right_menu')
+        self.sleep_small()
+
+    def click_on_role_delete_btn(self):
+        self.base_selenium.LOGGER.info('Press on the delete button')
+        self.base_selenium.click(element='roles_and_permissions:delete')
+        self.sleep_small()
+
+    def clear_role_name(self):
+        self.base_selenium.LOGGER.info('Clear role name')
+        self.base_selenium.clear_text(element='roles_and_permissions:role_name')
+
+    def click_on_pagination_page(self):
+        self.base_selenium.LOGGER.info('click on the pagination page')
+        self.base_selenium.click(element='roles_and_permissions:pagination_page')
+        self.sleep_small()
```

Comment:
We design the methods to match the end-user actions. There is no reason to create a method that just `click` an element. As a general role Method = User Action

Reply:
Done